var/spack/repos/builtin/packages/espresso/package.py

```diff
@@ -85,6 +88,14 @@ class Espresso(Package):
     conflicts('+elpa', when='@:5.4.0')
     conflicts('+hdf5', when='@:5.4.0')
 
+    # To build with QMCPACK's hdf5 interface we version 5.3.0
+    # and hdf5
+    conflicts('+qmcpackconv', when='@5.4:')
+    depends_on('hdf5+mpi', when='+qmcpackconv+mpi')
+    depends_on('hdf5~mpi', when='+qmcpackconv~mpi')
```

Comment:
You should also add `depends_on('hdf5', when='+qmcpackconv')`. Otherwise you can't do `spack install espresso +qmcpackconv ^hdf5@1.8.12`.